web/src/message_util.ts

```diff
@@ -24,13 +24,52 @@ export function do_unread_count_updates(messages: Message[], expect_no_new_unrea
     }
 }
 
-export function get_count_of_messages_in_topic_sent_after_current_message(
+export function get_count_of_messages_in_view_sent_after_current_message(
     stream_id: number,
     topic: string,
     message_id: number,
 ): number {
-    const all_messages = get_loaded_messages_in_topic(stream_id, topic);
-    return all_messages.filter((msg) => msg.id >= message_id).length;
+    let all_messages = get_messages_in_view_currently_focused(stream_id, topic);
```

Comment:
Don't call this `all_messages`! It's not "all" of anything natural. I'd call this

`const loaded_messages_in_current_view = ...`

and then the other block would be `const loaded_messages_in_all_messages` or something -- they're totally unrelated and the `let` here reusing the same variable name is also a bad pattern.